lua/barbar/render.lua

```diff
@@ -139,6 +139,28 @@ local function hl_tabline(group)
   return '%#' .. group .. '#'
 end
 
+-- Digits for optional styling of buffer_number and buffer_index.
+local SUPERSCRIPT_DIGITS = { '⁰', '¹', '²', '³', '⁴', '⁵', '⁶', '⁷', '⁸', '⁹' }
+local SUBSCRIPT_DIGITS = { '₀', '₁', '₂', '₃', '₄', '₅', '₆', '₇', '₈', '₉' }
+
+---@param num number
+---@param style barbar.config.options.icons.numbers.value
+local function style_number(num, style)
+  if style ~= 'superscript' and style ~= 'subscript' then
+    return tostring(num)
+  end
```

Comment:
So I know we've been using similar string constants for these, but I think we should use a named integer constants here, something like:

```lua
local BufferIndexVisibility = {
  HIDE = 0,
  SHOW = 1,
  SUPERSCRIPT = 2,
  SUBSCRIPT = 3,
}
```

And export it for users to use. I'm particularly not liking the string constants because this function is called on every render for every buffer in the list, and each of these `style == '...'` is a `strcmp(...)` to a string that needs to be stored & accessed on the heap, while integers are small efficient stack values :| @Iron-E wdyt?

Reply:
Are we sure string constants aren't implemented like `&str` in Rust (i.e. an array of `char`, and a stack value)? We could always do `local SUPERSCRIPT = 'superscript'` at the top too, but LuaJIT probably does stuff like that for constant values anyway. Dynamic strings are the ones I doubt LuaJIT can optimize

The real bottleneck is probably the `string.gmatch` down below. This is probably a better implementation of that:

```lua
  local digits
  local num_as_str = tostring(num)
  if style == 'subscript' then
    digits = SUBSCRIPT_DIGITS
  elseif style == 'superscript' then
    digits = SUPERSCRIPT_DIGITS
  else
    return num_string
  end

  local styled_number = ''
  for digit in string.gmatch(num_as_str, '%d') do
    styled_number = styled_number .. digits[digit + 1]
  end
```

Reply:
Yep, that is definitely a good suggestion. And please be both as picky on the PR! I'd love to it improved as much as possible :+1: